### PR TITLE
[primTorch] Add ref for `poisson_nll_loss`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #83805
* #83804
* #83803

### CODE REVIEW

**Comment by nkaretnikov** [comment]
@Lezcano had to re-run a ci job that failed mid-build, all green now. ptal

File: torch/_refs/nn/functional/__init__.py [hunk 1]
```diff
@@ -518,6 +519,43 @@ def gelu(a: TensorLikeType, approximate: str = "none") -> TensorLikeType:
...
+        reduction = _get_string_reduction_arg(size_average=size_average, reduce=reduce)
+    _check_reduction_value(reduction)
+    if log_input:
+        loss = torch.exp(input) - target * input
+    else:
+        loss = input - target * torch.log(input + eps)
+
+    if full:
+        stirling_term = (
+            target * torch.log(target) - target + 0.5 * torch.log(2 * torch.pi * target)
```
Comment (Lezcano) [inline torch/_refs/nn/functional/__init__.py:552]: Use `torch.xlogy` to get a value of 0 at target = 0.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -6990,6 +6990,26 @@ def gen_shape_kwargs():
     for input, target, kwargs in gen_shape_kwargs():
         yield SampleInput(input, args=(target, ), kwargs=kwargs)
 
+    # test INT_TO_FLOAT promotion
```
Comment (Lezcano) [inline torch/testing/_internal/common_methods_invocations.py:6993]: nice!

File: torch/_refs/nn/functional/__init__.py [hunk 3]
```diff
@@ -518,6 +519,43 @@ def gelu(a: TensorLikeType, approximate: str = "none") -> TensorLikeType:
...
+    """
+    if size_average is not None or reduce is not None:
+        # TODO: raise exception instead of converting value
+        # msg = "size_average and reduce args are deprecated, please use reduction argument."
+        reduction = _get_string_reduction_arg(size_average=size_average, reduce=reduce)
+    _check_reduction_value(reduction)
+    if log_input:
+        loss = torch.exp(input) - target * input
+    else:
+        loss = input - target * torch.log(input + eps)
```
Comment (Lezcano) [inline torch/_refs/nn/functional/__init__.py:548]: You can use `torch.xlogy` here.

**Review by lezcano — APPROVED** [review]
A couple minor things. If they make the tests error out just disregard them.

File: torch/_refs/nn/functional/__init__.py (hunk 1 above)
Comment (nkaretnikov) [reply torch/_refs/nn/functional/__init__.py:552]: no _refs support for `torch.xlogy`, so skipping

**Comment by nkaretnikov** [comment]
@ngimel a gentle reminder to look at the rest of this stack as the ci is green

File: torch/_refs/nn/functional/__init__.py (hunk 1 above)
Comment (ngimel) [reply torch/_refs/nn/functional/__init__.py:552]: It doesn't matter if there's refs support, you can still use it.

**Review by ngimel — APPROVED** [review]

File: torch/_refs/nn/functional/__init__.py (hunk 1 above)
Comment (nkaretnikov) [reply torch/_refs/nn/functional/__init__.py:552]: by ref support, i meant that the ref tests failed when i added it. so the current code seems better, no?

**Pull request closed without merging**